### ISSUE TITLE
Converting st names to directions

### ISSUE THREAD

**Issue by Pvp850** [slu-openGIS/postmastr#28]
**Before you open your issue:**
* All issues and contributions are covered by the [Code of Conduct](/.github/CODE_OF_CONDUCT.md)
* Please check out the [Contribution guidelines](/.github/CONTRIBUTING.md)

**Describe the bug**
Hello,
Thanks for this. Let me know if you need more testers. I also have preliminary code set up for apt number matching if you would like that. 

Anyways, back to the bug. It correctly converts North to N or East to E and even keeps directions in street addresses as directions, like E 28th st -> E 28th st. But what I found it does in error is convert letter street names to directions. I'm not sure why it does this, but this should not occur. For example:
E St -> East St. 
N Avenue -> North Ave
N N Avenue -> N North Ave

Should not occur because there are streets that are N and E streets. 

**Expected behavior**
It should keep letter names for sts. 
E St -> E St. 
N Avenue -> N Ave
N N Avenue -> N N Ave

**To Reproduce**
You can use the examples provided above.

```r

cx2 <- pm_identify(cx2, var = address)



##NEW DATAFRAME CX3 WITH PARSED VARIABLES FOR ADDRESS TO EXTRACT ADDRESS INTO ITS CONSTITUTENT PARTS
cx3 <- cx2 %>%
  pm_parse(input = "short", address = "address", output = "short",keep_parsed="yes",
           dir_dict = dirs, suffix_dict = suff )
```

**Screenshots**

**Desktop (please complete the following information):**
Windows 11
R 4.2.2
R studio 2021.09.0 Build 351

**Additional context**

**Comment by alankjackson** [comment]
Yes, I just ran into a similar problem. Street names like "Avenue N", "Avenue S", etc. fail. It obviously gets very confused. My workaround will be to protect those names and then repair them afterwards.
